src/boss/lambdastore/store.go

```diff
@@ -277,16 +309,17 @@ func (s *LambdaStore) getConfig(funcName string) (*common.LambdaConfig, error) {
 	lambdaEntry.Lock.Lock()
 	defer lambdaEntry.Lock.Unlock()
 
-	tarPath := filepath.Join(s.StorePath, funcName+".tar.gz")
-	cfg, err := common.ExtractConfigFromTarGz(tarPath)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, fmt.Errorf("lambda %q not found", funcName)
-		}
-		return nil, fmt.Errorf("failed to extract config: %w", err)
+	// Return cached config if available
+	if lambdaEntry.Config != nil {
+		return lambdaEntry.Config, nil
+	}
+
+	// If not cached, try to load from blob storage
+	if err := s.loadConfigAndRegister(funcName); err != nil {
+		return nil, fmt.Errorf("lambda %q not found", funcName)
```

Comment:
How do we know the err is not found, and not something else?  Probably better to propagate with %w.